plugins/rikki/heroeslounge/models/Division.php

```diff
@@ -85,12 +85,23 @@ public function getLongTitleAttribute()
     {
         if ($this->season != null) {
             return $this->season->title . ' - ' . $this->title;
-        } else if ($this->playoff != null && $this->playoff->season != null) {
-            return $this->playoff->season->title . ' - ' . $this->playoff->title;
+        } else if ($this->playoff != null) {
+            return $this->playoff->longTitle . ' - ' . $this->title;
         }
         return $this->title;
     }
 
+    public static function listDivisionsWithLongTitle()
+    {
+        $divisions = Division::all();
+        $list = [];
+
+        foreach ($divisions as $division) {
+            $list[$division->id] = $division->longTitle;
+        }
+        return $list;
```

Comment:
This can be done with a single line using the collection and map

Reply:
I attempted to use `map`, but I'm not quite sure how to get the output that I need. The following is the closest I got, but I require an array of key value pairs of `$divison->id => $division->longTitle` for attaching the correct division to a match.

```php
Division::all()->map(function ($division) {
    return [$division->id => $division->longTitle];
})>flatten()->toArray();

Output:
  0 => 'Season 4 - Division 1',
  1 => 'Season 4 - Division 2',
  2 => 'Season 4 - Division 3',
...

Desired output:
  1 => 'Season 4 - Division 1',
  2 => 'Season 4 - Division 2',
  3 => 'Season 4 - Division 3',
...
```

Advice how to accomplish this is appreciated.

Reply:
I think the problem in your current solution is that you at first create small arrays and then use flatten which causes the keys to be enumerated. Maybe try 
`Division::all()->map(function ($division) {
    return $division->id => $division->longTitle;
})->toArray();`

Reply:
Better languages can solve this problem the way you describe, but php is far from that good.  As written, this really is the best way in php.  Here is some discussion of the options:
https://stackoverflow.com/questions/11563119/how-to-convert-an-array-of-arrays-or-objects-to-an-associative-array

Reply:
Oh although I dug a bit deeper and it turns out you can do it with array map if you jump through some extra hoops:
https://ryanwinchester.ca/posts/php-array-map-with-keys

Look at example #3 from the official docs:
https://www.php.net/manual/en/function.array-map.php

Reply:
Did you have a look at https://octobercms.com/docs/services/collections
Which is what October uses in many places instead of a bare PHP array

Reply:
Figured it out by using `collection->keyBy()` before `collection->map()`.